src/Core/Echo.Ast/Construction/AstParserContext.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using Echo.ControlFlow;
+using Echo.Core.Code;
+using Echo.DataFlow;
+
+namespace Echo.Ast.Construction
+{
+    internal sealed class AstParserContext<TInstruction>
+    {
+        private readonly ControlFlowGraph<TInstruction> _controlFlowGraph;
+
+        internal AstParserContext(
+            ControlFlowGraph<TInstruction> controlFlowGraph,
+            DataFlowGraph<TInstruction> dataFlowGraph)
+        {
+            _controlFlowGraph = controlFlowGraph;
```

Comment:
null check

Reply:
`AstParserContext` is internal so the parameters will always be checked for `null` before.